mutablesecurity/solutions/implementations/duplicati/code.py

```diff
@@ -0,0 +1,693 @@
+                f" {IDENTIFIER} | cut -d : -f 2"
+            )
+
+    IDENTIFIER = "passphrase"
+    DESCRIPTION = "This value represents the value by encryption key"
+    INFO_TYPE = StringDataType
+    PROPERTIES = [
+        InformationProperties.CONFIGURATION,
+        InformationProperties.OPTIONAL,
+        InformationProperties.WITH_DEFAULT_VALUE,
+        InformationProperties.NON_DEDUCTIBLE,
+        InformationProperties.WRITABLE,
+    ]
+    DEFAULT_VALUE = "mutablesecutiy"
+    GETTER = PassphraseValue
+    SETTER = set_configuration
+
+
+class RestorePath(BaseInformation):
+    IDENTIFIER = "restore_path"
+    DESCRIPTION = "Path of directory you want to restore your backup"
+    INFO_TYPE = StringDataType
+    PROPERTIES = [
+        InformationProperties.CONFIGURATION,
+        InformationProperties.OPTIONAL,
+        InformationProperties.WITH_DEFAULT_VALUE,
+        InformationProperties.NON_DEDUCTIBLE,
+        InformationProperties.WRITABLE,
+    ]
+    DEFAULT_VALUE = "/opt/mutablesecurity/duplicati/restorePath"
+    GETTER = None
+    SETTER = None
+
+
+# Logs classes definitions
+class DefaultLogs(BaseLog):
+    class DefaultLogsFact(FactBase):
+        command = "cat /var/log/duplicati.log"
+
+        @staticmethod
+        def process(output: typing.List[str]) -> str:
+            return "\n".join(output)
+
+    IDENTIFIER = "logs"
+    DESCRIPTION = "Default log location"
+    FACT = DefaultLogsFact
+
+
+# Tests
+class SupportedArchitecture(BaseTest):
+    class SupportedArchitectureFact(BinaryArchitectureFact):
+        @staticmethod
+        def process(  # type: ignore[override]
+            output: typing.List[str],
+        ) -> bool:
+            architecture = BinaryArchitectureFact.process(output)
+
+            return architecture is not None
+
+    IDENTIFIER = "supported_architecture"
+    DESCRIPTION = "Checks if there is any build for this architecture."
+    TEST_TYPE = TestType.REQUIREMENT
+    FACT = SupportedArchitectureFact
+
+
+class ClientCommandPresence(BaseTest):
+    IDENTIFIER = "command"
+    DESCRIPTION = "Checks if the Duplici cli is registered as a command."
+    TEST_TYPE = TestType.PRESENCE
+    FACT = PresentCommand
+    FACT_ARGS = ("duplicati-cli --help",)
+
+
+class ClientEncryption(BaseTest):
+    class EncryptionTest(FactBase):
+        @staticmethod
+        def command() -> str:
+            backup_path = "/tmp/backup" + uuid.uuid4().hex
+            file_test = "/tmp/file.txt"
+            execute_command = [
+                "echo 'Hi' >> f{file_test}",
+                DuplicatiHelper._make_local_backup(file_test, backup_path)
+                + f" && dir {backup_path} |grep '*.f{EncryptionModule.get()}'",
+                f"rm -r {backup_path}",
+            ]
+
+            return execute_command
+
+        @staticmethod
+        def process(output: typing.List[str]) -> bool:
+            return int(output[0]) != 0
+
+    IDENTIFIER = "encryption_test"
```

Comment:
"encryption_test" may be too generic. Let's try to make it more specific (maybe "local_encrypted_backup", just an idea).

Reply:
done